src/lib.rs

```diff
@@ -227,6 +227,33 @@ impl Sample for i32 {
     }
 }
 
+impl Sample for f32 {
+    fn write<W: io::Write>(self, writer: &mut W, bits: u16) -> Result<()> {
+        match bits {
+            32 => Ok(try!(writer.write_le_f32(self))),
+            _ => Err(Error::Unsupported),
+        }
+    }
+
+    fn read<R: io::Read>(reader: &mut R, bytes: u16, bits: u16) -> Result<Self> {
+        match (bytes, bits) {
+            (4, 32) => Ok(try!(reader.read_le_f32())),
+            (n, _) if n > 4 => Err(Error::TooWide),
+            _ => Err(Error::Unsupported),
+        }
+    }
+}
+
+/// Specifies whether a sample is stored as an "IEEE Float" or an integer.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SampleFormat {
+    /// Wave files with the `WAVE_FORMAT_IEEE_FLOAT` format tag store samples as floating point
+    /// values with a full scale of 1.0.
```

Comment:
Ah, after reading [“Non-PCM Formats”](www-mmsp.ece.mcgill.ca/documents/audioformats/wave/wave.html) this makes sense. I’d change it to “Values are normally in the range [-1.0, 1.0].” to clarify that it is a convention, not a guarantee.